storage/shard.go

```diff
@@ -296,27 +303,38 @@ func (s *dbShard) Bootstrap(bs m3db.Bootstrap, writeStart time.Time, cutover tim
 	// Find the series with no data within the retention period but has
 	// buffered data points since server start. Any new series added
 	// after this will be marked as bootstrapped.
-	var bufferedSeries []databaseSeries
-	s.RLock()
-	for id, entry := range s.lookup {
-		if _, exists := bootstrappedSeries[id]; !exists {
-			bufferedSeries = append(bufferedSeries, entry.series)
+	if bootstrappedSeries == nil {
+		// If we failed to bootstrap any series from the external sources,
+		// we need to bootstrap all the series seen so far.
+		s.forEachSeries(true, func(series databaseSeries) error {
+			err := series.Bootstrap(nil, cutover)
+			multiErr = multiErr.Add(err)
+			return err
+		})
+	} else {
+		// Otherwise we only bootstrap the series we haven't bootstrapped
+		// from external sources.
+		var bufferedSeries []databaseSeries
+		s.RLock()
+		for id, entry := range s.lookup {
+			if _, exists := bootstrappedSeries[id]; !exists {
+				bufferedSeries = append(bufferedSeries, entry.series)
+			}
 		}
-	}
-	s.RUnlock()
+		s.RUnlock()
 
-	// Finally bootstrapping series with no recent data.
-	for _, series := range bufferedSeries {
-		if err := series.Bootstrap(nil, cutover); err != nil {
-			return err
+		// Finally bootstrapping series with no recent data.
+		for _, series := range bufferedSeries {
+			err := series.Bootstrap(nil, cutover)
+			multiErr = multiErr.Add(err)
 		}
 	}
```

Comment:
yea so i was hoping to avoid repeatedly doing nil checks against `bootstrappedSeries` for each series, and more importantly, avoid holding the lock when we can during series bootstrapping since that could be a lengthy operation.